etc/kayobe/kolla/inventory/group_vars/prometheus-blackbox-exporter

```diff
@@ -6,13 +6,16 @@
 # prometheus_blackbox_exporter_endpoints_kayobe is another set of default
 # endpoints that are templated by Kayobe rather than Kolla Ansible. See
 # kolla/globals.yml for more details.
-prometheus_blackbox_exporter_endpoints: >-
-  {{ (prometheus_blackbox_exporter_endpoints_kayobe | default([]) +
-  prometheus_blackbox_exporter_endpoints_default) |
-  selectattr('enabled', 'true') |
-  map(attribute='endpoints') | flatten |
-  union(prometheus_blackbox_exporter_endpoints_custom) |
-  unique | select | list }}
+prometheus_blackbox_exporter_endpoints: |
+  {% set endpoints = [] %}
+  {% for dict_item in (prometheus_blackbox_exporter_endpoints_kayobe | default([]) + prometheus_blackbox_exporter_endpoints_default) %}
+  {% if dict_item.enabled | bool %}
+  {% for endpoint in dict_item.endpoints %}
+  {% set _ = endpoints.append(endpoint) %}
+  {% endfor %}
+  {% endif %}
+  {% endfor %}
+  {{ (endpoints + prometheus_blackbox_exporter_endpoints_custom) | unique | list }}
```

Comment:
There is a `select` in the original filter pipeline. Is it possible that an endpoint could be set to an empty string?

Reply:
I think if `kolla_same_external_internal_vip` is true, then there will be a lot of empty string endpoints, so select would probably be quite important

Reply:
Done